MapboxCoreNavigation/NavigationSettings.swift

```diff
@@ -23,10 +23,28 @@ public class NavigationSettings: NSObject {
     @objc public dynamic var voiceVolume    : Float     = 1.0
     
     /**
-     Indicates whether the voice controller should be muted or not.
+     Specifies whether to mute the voice controller or not.
      */
     @objc public dynamic var voiceMuted     : Bool      = false
     
+    /**
+     Specifies the preferred distance measurement unit.
+     - note: Anything but `kilometer` and `mile` will fall back to the default measurement for the current locale.
+        Meters and feets will be used when the presented distances are small enough. See `DistanceFormatter` for more information.
+     */
+    @objc public dynamic var distanceUnit : LengthFormatter.Unit = Locale.current.usesMetric ? .kilometer : .mile
```

Comment:
Once the Directions API supports yards for the UK (see also mapbox/mapbox-navigation-ios#242), `mile` will become ambiguous. Consider defining a new enumeration along the lines of MapKit’s `MKDistanceFormatterUnits`.

Reply:
Kilometers or miles seems to be a consensus in all navigation and fitness apps I've seen. However, should be fairly straightforward to support yards for walking/cycling in the UK as tail work if needed.

Reply:
The UK uses yards on [a number of road signs](https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/203668/traffic-signs-manual-chapter-07.pdf#page=155), so it’d be prudent to match signage. This is also the stance [CLDR](https://unicode.org/cldr/trac/browser/trunk/common/supplemental/supplementalData.xml?rev=13744#L4833) has taken, so iOS follows this convention.

Reply:
Figuring out when to use yards would be an additional feature that we'd have to add support for cases like a sign indicating railway station, ferry, etc. The distance unit would still be set to miles, but the distance formatter would have to become smarter to figure out when to display yards or feet.

Reply:
The heuristic would be pretty simple: when LengthFormatter wants to use yards, use yards instead of forcing feet, as we do now. Am I missing something?

Reply:
> Am I missing something?

It's probably me, IDK, U.K is confusing. I was under the impression that U.K only uses yards for some special edge cases or that we regressed but that doesn't seem to be the case.

Is https://github.com/mapbox/mapbox-navigation-ios/pull/1055/commits/9c42982799a2f2c7fef3e57a487471c34379bd72 looking better?